packages/account/src/Containers/Account/account.tsx

```diff
@@ -31,16 +31,30 @@ const Account = observer(({ history, location, routes }: TAccountProps) => {
         should_allow_authentication,
         should_allow_poinc_authentication,
     } = client;
-    const { toggleAccountSettings, is_account_settings_visible } = ui;
+    const { toggleAccountSettings, is_account_settings_visible, is_mobile, is_desktop } = ui;
+
+    const [available_routes, setAvailableRoutes] = React.useState(routes);
+    const { is_passkey_supported, is_passkey_support_checking } = useIsPasskeySupported();
+
+    const should_remove_passkeys_route = is_desktop || (is_mobile && !is_passkey_supported);
+
+    React.useEffect(() => {
+        if (is_passkey_support_checking) return;
+        if (should_remove_passkeys_route) {
+            const desktop_routes = removeExactRouteFromRoutes(routes, 'passkeys');
+            setAvailableRoutes(desktop_routes as TRoute[]);
```

Comment:
```suggestion
            setAvailableRoutes(desktop_routes);
```

Reply:
can't do it because of different types for account and for shared :(